### DISCUSSION TITLE
TODOMVC in Pynecone?

### DISCUSSION THREAD

**Discussion by metaperl** [reflex-dev/reflex#1018] [Q&A]
I would be curious to see [TODOMVC](https://todomvc.com) implemented in Pynecone.

I have already added [Solara's implementation of TODOMVC](https://github.com/metaperl/pure-python-web-development/tree/main/examples/todomvc/solara) to the Pure Python Web Dev survey repo.

Would love to add a Pynecone example ;)

**Comment by Alek99** [comment]
We have simple todo app example here https://pynecone.io/docs/library/layout/foreach feel free to modify it to fit the spec, seems to be pretty similar with a couple more features like editing todo

**Reply by metaperl** [reply]
Regarding this example:
1. it does not clear the entered text when you press 'Add'
2. I am concerned that the TODO items are not pure model objects, but instead inherit from `pc.State`. I think one of the points of TODOMVC was to show how cleanly a framework separates model from view (and controller).
